### PR TITLE
Update plugins core 0.6.x

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java [hunk 1]
```diff
@@ -216,31 +150,69 @@ public GenRepoInfoFile genRepoInfoFile() {
...
-    if (mojo.getCloudSdkHome() != null && mojo.getCloudSdkVersion() != null) {
+  private AppCfg getAppCfg() {
+    return AppCfg.builder(cloudSdk).build();
+  }
+
+  private LocalRun getLocalRun() {
+    return LocalRun.builder(cloudSdk).build();
+  }
+
+  ProcessHandler newDefaultProcessHandler() {
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java]: private?

File: src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java [hunk 2]
```diff
@@ -216,31 +150,69 @@ public GenRepoInfoFile genRepoInfoFile() {
...
+    ProcessOutputLineListener lineListener = new DefaultProcessOutputLineListener(mojo.getLog());
+    return LegacyProcessHandler.builder()
+        .addStdOutLineListener(lineListener)
+        .addStdErrLineListener(lineListener)
+        .setExitListener(new NonZeroExceptionExitListener())
+        .build();
+  }
+
+  ProcessHandler newDevAppServerAsyncHandler(int timeout) {
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactory.java]: private?

File: src/main/java/com/google/cloud/tools/maven/CloudSdkLoginMojo.java [hunk 3]
```diff
@@ -17,21 +17,19 @@
...
-import org.apache.maven.plugin.MojoExecutionException;
-import org.apache.maven.plugin.MojoFailureException;
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdkNotFoundException;
 import org.apache.maven.plugins.annotations.Mojo;
 
 /** Login and set the Cloud SDK common configuration user. */
 @Mojo(name = "cloudSdkLogin")
 public class CloudSdkLoginMojo extends CloudSdkMojo {
 
+  protected CloudSdkLoginMojo() throws CloudSdkNotFoundException {}
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/maven/CloudSdkLoginMojo.java]: Is this used?

File: src/test/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactoryTest.java [hunk 4]
```diff
@@ -94,130 +88,26 @@ public void wireUp()
...
-    factory.devServerRunSync(SupportedDevServerVersion.V1);
-
-    // verify
-    verify(cloudSdkBuilderMock).build();
-    verify(cloudSdkFactoryMock).devServer1(cloudSdkMock);
-    verifyDefaultCloudSdkBuilder();
-  }
-
-  @Test
-  public void testDevServer2RunSync() throws CloudSdkNotFoundException {
```
Comment (coollog) [inline src/test/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactoryTest.java]: Are these run/stop tests not relevant anymore?

Comment (loosebazooka) [reply src/test/java/com/google/cloud/tools/maven/CloudSdkAppEngineFactoryTest.java]: not really, they never did anything, and the underlying mechanism is different now.

File: src/main/java/com/google/cloud/tools/maven/CloudSdkLoginMojo.java (hunk 3 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/maven/CloudSdkLoginMojo.java]: ah yeah, need to get rid of this.

**Pull request merged**